base/applications/rapps/include/rosui.h

```diff
@@ -374,173 +396,150 @@ class CUiPanel :
         m_Height = CUiMeasure::FitParent();
     }
 
-    virtual ~CUiPanel()
-    {
-    }
+    virtual ~CUiPanel() {}
 
-    virtual CUiBox * AsBox() { return this; }
+    // virtual CUiBox* AsBox() { return this; }
 
     virtual VOID ComputeMinimalSize(SIZE* size)
     {
         for (INT i = 0; i < m_Children.GetCount(); i++)
         {
-            CUiBox * box = m_Children.Get(i)->AsBox();
```

Comment:
Here (and elsewhere) I have the impression that this indirection looks quite unnecessary.

Indeed: why would we have elements in the children collection of `CUiPrimitive`s that would not be measurable UI boxes? In what sort of situations this could happen? (I'm asking in a generic manner, not in regards to the current usage of this UI library by RAPPS). What sort of so-called `CUiPrimitive`s could be added into the children collection, yet not have a box? Would this make any sense?
Cc @gigaherz for any clues...